bftengine/src/bftengine/SeqNumInfo.cpp

```diff
@@ -430,29 +301,24 @@ PrepareFullMsg* SeqNumInfo::ExFuncForPrepareCollector::createCombinedSignatureMs
       viewNumber, seqNumber, r->getReplicasInfo().myId(), combinedSig, combinedSigLen, span_context);
 }
 
-InternalMessage* SeqNumInfo::ExFuncForPrepareCollector::createInterCombinedSigFailed(
-    void* context, SeqNum seqNumber, ViewNum viewNumber, std::set<uint16_t> replicasWithBadSigs) {
-  InternalReplicaApi* r = (InternalReplicaApi*)context;
-  return new CombinedSigFailedInternalMsg(r, seqNumber, viewNumber, replicasWithBadSigs);
+InternalMessage SeqNumInfo::ExFuncForPrepareCollector::createInterCombinedSigFailed(
+    SeqNum seqNumber, ViewNum viewNumber, std::set<uint16_t> replicasWithBadSigs) {
+  return CombinedSigFailedInternalMsg(seqNumber, viewNumber, replicasWithBadSigs);
 }
 
-InternalMessage* SeqNumInfo::ExFuncForPrepareCollector::createInterCombinedSigSucceeded(
-    void* context,
+InternalMessage SeqNumInfo::ExFuncForPrepareCollector::createInterCombinedSigSucceeded(
     SeqNum seqNumber,
     ViewNum viewNumber,
     const char* combinedSig,
     uint16_t combinedSigLen,
     const std::string& span_context) {
-  InternalReplicaApi* r = (InternalReplicaApi*)context;
-  return new CombinedSigSucceededInternalMsg(r, seqNumber, viewNumber, combinedSig, combinedSigLen, span_context);
+  return CombinedSigSucceededInternalMsg(seqNumber, viewNumber, combinedSig, combinedSigLen, span_context);
 }
 
-InternalMessage* SeqNumInfo::ExFuncForPrepareCollector::createInterVerifyCombinedSigResult(void* context,
-                                                                                           SeqNum seqNumber,
-                                                                                           ViewNum viewNumber,
-                                                                                           bool isValid) {
-  InternalReplicaApi* r = (InternalReplicaApi*)context;
-  return new VerifyCombinedSigResultInternalMsg(r, seqNumber, viewNumber, isValid);
+InternalMessage SeqNumInfo::ExFuncForPrepareCollector::createInterVerifyCombinedSigResult(SeqNum seqNumber,
+                                                                                          ViewNum viewNumber,
+                                                                                          bool isValid) {
+  return VerifyCombinedSigResultInternalMsg(seqNumber, viewNumber, isValid);
```

Comment:
Why do we have those `create*` functions in the first place? Maybe another potential refactoring task for later?

Reply:
I had the same thought at first, and actually started making that change, but it doesn't work unfortunately. Although, this possibly can be made simpler with another refactoring. The reason we can't do this is that [CollectorOfThresholdSignatures](https://github.com/vmware/concord-bft/blob/master/bftengine/src/bftengine/CollectorOfThresholdSignatures.hpp#L31) uses these wrappers for both prepare and commit messages which are different.